packages/flutter/lib/src/rendering/view.dart

```diff
@@ -184,6 +204,35 @@ class RenderView extends RenderObject with RenderObjectWithChildMixin<RenderBox>
     }
   }
 
+  void _updateSystemChrome() {
+    if (!automaticSystemUiAdjustment)
+      return;
+    final Rect bounds = paintBounds;
+    final Offset top = new Offset(bounds.center.dx, ui.window.padding.top / ui.window.devicePixelRatio);
```

Comment:
why is the top coordinate that you use at the bottom of the bar, but the bottom color that you use at the bottom of its bar? it seems weirdly asymmetric.

Reply:
I think this is what you meant ?